src/z3c/autoinclude/utils.py

```diff
@@ -151,17 +165,22 @@ def find_packages(where='.', exclude=()):
     names, such that 'foo.*' will exclude all subpackages of 'foo' (but not
     'foo' itself).
     """
+    original_where = where
+    if where in CACHE:
+        return CACHE[where]
     out = []
     stack=[(convert_path(where), '')]
     while stack:
         where,prefix = stack.pop(0)
+        os.listdir(where)
         for name in os.listdir(where):
             fn = os.path.join(where,name)
-            if ('.' not in name and os.path.isdir(fn) and
-                os.path.isfile(os.path.join(fn,'__init__.py'))
-            ):
-                out.append(prefix+name); stack.append((fn,prefix+name+'.'))
+            if ('.' not in name and os.path.isdir(fn)):
+                stack.append((fn, prefix+name+'.'))
+                if os.path.isfile(os.path.join(fn, '__init__.py')):
+                    out.append(prefix+name)
```

Comment:
hmm, reading the pep, I could maybe just use this: https://docs.python.org/2/library/pkgutil.html#pkgutil.walk_packages